components/General/PrivateRoute.js

```diff
@@ -52,9 +54,34 @@ export default class PrivateRoute extends React.PureComponent {
         // NOTE: props contain params to be sent to child only
 
         if (PrivateRoute.shouldRedirect(authenticated, invertBehavior)) {
+            const from = props.location || PrivateRoute.defaultLocation;
+            let search;
+
+            if (authenticated) {
```

Comment:
Shouldn't we have a check if invertBehavior is true and ignore that case altogether?